### ISSUE TITLE
Add some docs about how to set up local dev env

### ISSUE THREAD

**Issue by yangwucheng** [4paradigm/OpenMLDB#3758]
**Describe the feature you'd like**

目前整个代码库项目比较多，依赖语言和环境也比较复杂，通过目前的文档可以较为顺利地安装和部署，但是如果需要进行本地调试开发，相关编译调试文档还是比较欠缺的，希望可以补充相关文档降低开发环境配置门槛。

**Additional context**

目前找不到相关文档阐述开发环境编译配置，例如：
1. openmldb项目的`mvn compile` 等操作之前需要先进行 `make` 操作，
2. openmldb项目的 `mvn install` 等操作会触发 gpg, scalatest 等插件运行，在本地执行一般会失败，需要skip相关插件，可以配置`install -Dgpg.skip -DskipTests=true -Dwagon.skip=true -Dmaven.test.skip=true -f pom.xml`
3. 3.8.1及以后的maven不支持http repository，项目使用的 confluent repo(https://packages.confluent.io/maven) 中相关库会出现下载失败等情况，可以使用以下方式解决：
  3.1 maven降级
  3.2 maven settings.xml 中增加 mirror 配置
```xml
    <mirror>
            <id>maven-default-http-blocker</id>
            <url>http://127.0.0.1/dont-go-here</url>
            <mirrorOf>dummy</mirrorOf>
            <blocked>false</blocked>
        </mirror>
```

**Comment by aceforeverd** [comment]
> 目前找不到相关文档阐述开发环境编译配置，例如：
> 
> 1. openmldb项目的`mvn compile` 等操作之前需要先进行 `make` 操作，
> 2. openmldb项目的 `mvn install` 等操作会触发 gpg, scalatest 等插件运行，在本地执行一般会失败，需要skip相关插件，可以配置`install -Dgpg.skip -DskipTests=true -Dwagon.skip=true -Dmaven.test.skip=true -f pom.xml`

agree, we may add README in java directory, or jdk related docs in docs directory

> 3. 3.8.1及以后的maven不支持http repository，项目使用的 confluent repo(https://packages.confluent.io/maven) 中相关库会出现下载失败等情况，可以使用以下方式解决：
>    3.1 maven降级
>    3.2 maven settings.xml 中增加 mirror 配置
> 
> ```
>     <mirror>
>             <id>maven-default-http-blocker</id>
>             <url>http://127.0.0.1/dont-go-here</url>
>             <mirrorOf>dummy</mirrorOf>
>             <blocked>false</blocked>
>         </mirror>
> ```

you should use the builtin mvn wrapper in java directory

**Comment by vagetablechicken** [comment]
https://openmldb.ai/docs/zh/main/developer/sdk_develop.html

You can add in here, rename it as java dev. And scalatest bug will be fixed by #3755, no need to add in doc.

**Comment by vagetablechicken** [comment]
> 
> 
> > 目前找不到相关文档阐述开发环境编译配置，例如：
> 
> > 
> 
> > 1. openmldb项目的`mvn compile` 等操作之前需要先进行 `make` 操作，
> 
> > 2. openmldb项目的 `mvn install` 等操作会触发 gpg, scalatest 等插件运行，在本地执行一般会失败，需要skip相关插件，可以配置`install -Dgpg.skip -DskipTests=true -Dwagon.skip=true -Dmaven.test.skip=true -f pom.xml`
> 
> 
> 
> agree, we may add README in java directory, or jdk related docs in docs directory
> 
> 
> 
> > 3. 3.8.1及以后的maven不支持http repository，项目使用的 confluent repo(https://packages.confluent.io/maven) 中相关库会出现下载失败等情况，可以使用以下方式解决：
> 
> >    3.1 maven降级
> 
> >    3.2 maven settings.xml 中增加 mirror 配置
> 
> > 
> 
> > ```
> 
> >     <mirror>
> 
> >             <id>maven-default-http-blocker</id>
> 
> >             <url>http://127.0.0.1/dont-go-here</url>
> 
> >             <mirrorOf>dummy</mirrorOf>
> 
> >             <blocked>false</blocked>
> 
> >         </mirror>
> 
> > ```
> 
> 
> 
> you should use the builtin mvn wrapper in java directory 
> 
> 

He used IDEA, we can give a hint about IDEA use project wrapper, or  developer should use the right version

**Comment by vagetablechicken** [comment]
aliyun maven may lost some packages, developer should know it.
